packages/@react-aria/utils/src/useSyncRef.ts

```diff
@@ -23,7 +23,9 @@ export function useSyncRef<T>(context?: ContextValue<T> | null, ref?: RefObject<
     if (context && context.ref && ref) {
       context.ref.current = ref.current;
       return () => {
-        context.ref.current = null;
+        if (context.ref) {
+          context.ref.current = null;
```

Comment:
What's the reason for this making this an `if` check instead of something like `context?.ref?.current = null;`?

Reply:
you can't assign to an optional chain